packages/core/src/sync-realtime/index.ts

```diff
@@ -587,19 +683,38 @@ export const createRealtimeSync = (
     });
 
     // Remove call traces that don't match a filter, recording required transactions
-    callTraces = callTraces.filter((callTrace) => {
-      let isCallTraceMatched = false;
-      for (const filter of callTraceFilters) {
-        if (isCallTraceFilterMatched({ filter, block, callTrace })) {
-          isCallTraceMatched = true;
-          requiredTransactions.add(callTrace.transactionHash);
-          if (filter.includeTransactionReceipts) {
-            requiredTransactionReceipts.add(callTrace.transactionHash);
-          }
+    // callTraces = callTraces.filter((callTrace) => {
+    //   let isCallTraceMatched = false;
+    //   for (const filter of callTraceFilters) {
+    //     if (isCallTraceFilterMatched({ filter, block, callTrace })) {
+    //       isCallTraceMatched = true;
+    //       requiredTransactions.add(callTrace.transactionHash);
+    //       if (filter.includeTransactionReceipts) {
+    //         requiredTransactionReceipts.add(callTrace.transactionHash);
+    //       }
+    //     }
+    //   }
+
+    //   return isCallTraceMatched;
+    // });
+
+    // Remove traces that dont match transaction or transfer filter
+    traces = traces.filter((trace) => {
```

Comment:
I think we need to do a depth first search over all the traces, either in this block or above where `traces` is declared. 

It performs three functions:
- processes child traces (each call can have several sub-calls)
- orders traces for `checkpoint.eventIndex`
- determine failed traces

See `sync-historical` https://github.com/ponder-sh/ponder/blob/2117cec2f8e477d5446ab0242b07a82d054f50e1/packages/core/src/sync-historical/index.ts#L324-L368